### PR TITLE
Dev server api endpoints

### PR DESCRIPTION
**Pull request by Cobus-Redelinghuys**
Api endpoints brought up to rest standard
Unit tests moved into separate files
All tests fixed such that they will compile and run
Added ApiResponse object as clients requested
Removed some redundant request and response objects

### CODE REVIEW

File: Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/messenger/controllers/MessengerController.kt [hunk 1]
```diff
@@ -4,29 +4,31 @@ import com.savannasolutions.SmartContractVerifierServer.messenger.requests.*
...
 
 @CrossOrigin
 @RestController
-@RequestMapping("/messenger")
 class MessengerController constructor(private val messengerService: MessengerService) {
 
-    @PostMapping("/get-all-messages-by-agreement")
-    fun getAllMessagesByAgreement(@RequestBody getAllMessagesByAgreementRequest: GetAllMessagesByAgreementRequest) =
-        messengerService.getAllMessagesByAgreement(getAllMessagesByAgreementRequest)
+    @GetMapping("/user/{userId}/agreement/{agreementId}/message")
```
Comment (Kaleb-Bruwer) [inline Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/messenger/controllers/MessengerController.kt:13]: This definitely looks more elegant

File: Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/messenger/services/MessengerService.kt [hunk 2]
```diff
@@ -96,34 +96,55 @@ class MessengerService constructor(val messagesRepository: MessagesRepository,
...
 
-        if(!agreementsRepository.existsById(sendMessageRequest.AgreementID))
-            return SendMessageResponse(status = ResponseStatus.FAILED)
+        if(!agreementsRepository.existsById(agreementID))
+            return ApiResponse(status = ResponseStatus.FAILED,
+                        message = commonResponseErrorMessages.agreementDoesNotExist)
 
         if(sendMessageRequest.Message.isEmpty())
-            return SendMessageResponse(status = ResponseStatus.FAILED)
+            return ApiResponse(status = ResponseStatus.FAILED, message = "Message is empty")
```
Comment (Kaleb-Bruwer) [inline Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/messenger/services/MessengerService.kt:144]: Didn't you create common error messages?

File: Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/negotiation/requests/CreateAgreementRequest.kt [hunk 3]
```diff
@@ -2,8 +2,7 @@ package com.savannasolutions.SmartContractVerifierServer.negotiation.requests
 
 import com.fasterxml.jackson.annotation.JsonProperty
 
-data class CreateAgreementRequest(@JsonProperty("PartyA") val PartyA:String,
-                                  @JsonProperty("PartyB") val PartyB:String,
+data class CreateAgreementRequest(@JsonProperty("PartyB") val PartyB:String,
```
Comment (Kaleb-Bruwer) [inline Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/negotiation/requests/CreateAgreementRequest.kt:5]: Is PartyA assumed to be the logged in user?

**Review by Kaleb-Bruwer — COMMENTED** [review]
I see you have an empty file named negotiation/services/MergeClearer.txt, what's that for?
But overall it looks good, for the tests I only did a few spot checks.

File: Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/messenger/services/MessengerService.kt (hunk 2 above)
Comment (Cobus-Redelinghuys) [reply Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/messenger/services/MessengerService.kt:144]: So this is a new feature. In stead of just saying failed or successful we can now actually see what failed. I do have a class with common messages, but thus is an uncommon error message

File: Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/negotiation/requests/CreateAgreementRequest.kt (hunk 3 above)
Comment (Cobus-Redelinghuys) [reply Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/negotiation/requests/CreateAgreementRequest.kt:5]: Yes it is. Or well the user sending the request

File: Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/messenger/services/MessengerService.kt (hunk 2 above)
Comment (Kaleb-Bruwer) [reply Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/messenger/services/MessengerService.kt:144]: All right

**Review by Kaleb-Bruwer — APPROVED** [review]

**Pull request merged**